pyglmnet/pyglmnet.py

```diff
@@ -26,40 +28,39 @@ def _qu(distr, z, eta):
         qu[z <= eta] = np.exp(z[z <= eta])
     elif distr == 'gaussian':
         qu = z
+        grad_qu = np.ones_like(z)
     elif distr == 'binomial':
         qu = expit(z)
+        grad_qu = expit(z) * (1 - expit(z))
     elif distr == 'probit':
-        qu = probit(z)
-    elif distr == 'gamma':
-        qu = np.log1p(np.exp(z))
-    return qu
+        qu = norm.cdf(z)
+        grad_qu = norm.pdf(z)
+    if return_grad:  # eventually this if conditional should go away
+        return qu, grad_qu
+    else:
+        return qu
 
 
-def _logL(distr, beta0, beta, X, y, eta):
+def _logL(distr, y, y_hat):
     """The log likelihood."""
-    z = beta0 + np.dot(X, beta)
-    l = _qu(distr, z, eta)
-    if distr == 'softplus':
-        logL = np.sum(y * np.log(l) - l)
-    elif distr == 'poisson':
-        logL = np.sum(y * np.log(l) - l)
+    if distr in ['softplus', 'poisson']:
+        logL = np.sum(y * np.log(y_hat) - y_hat)
     elif distr == 'gaussian':
-        logL = -0.5 * np.sum((y - l)**2)
+        logL = -0.5 * np.sum((y - y_hat)**2)
     elif distr == 'binomial':
         # analytical formula
-        # logL = np.sum(y*np.log(l) + (1-y)*np.log(1-l))
+        logL = np.sum(y * np.log(y_hat) + (1 - y) * np.log(1 - y_hat))
 
         # but this prevents underflow
-        z = beta0 + np.dot(X, beta)
-        logL = np.sum(y * z - np.log(1 + np.exp(z)))
+        # z = beta0 + np.dot(X, beta)
+        # logL = np.sum(y * z - np.log(1 + np.exp(z)))
```

Comment:
can we leave these lines commented in? 

when i was first implementing it, i found some strange scenarios when calculating binomial log likelihood in the naive way. this is related to taking log of sigmoid. if you do the algebra you see `np.log(np.exp(z) / (1 + np.exp(z))` which is `z - np.log(1 + np.exp(z)`. i'm concerned the tests are passing because of a very liberal tolerance.